servers/rendering/renderer_rd/effects/ss_effects.cpp

```diff
@@ -1285,9 +1285,7 @@ void SSEffects::generate_ssao(Ref<RenderSceneBuffersRD> p_render_buffers, SSAORe
 				RD::get_singleton()->compute_list_dispatch_threads(compute_list, p_ssao_buffers.buffer_width, p_ssao_buffers.buffer_height, 1);
 			}
 
-			if (ssao_quality > RS::ENV_SSAO_QUALITY_VERY_LOW) {
-				RD::get_singleton()->compute_list_add_barrier(compute_list);
-			}
+			RD::get_singleton()->compute_list_add_barrier(compute_list);
```

Comment:
I don't think we want to be adding a barrier between every pass. Instead, this should likely be the same as SSIL and insert the barrier when `i == 3`

Reply:
Very low quality only does one pass, so it seemed redundant to add the if. Do you think we should add one for clarity anyways?

Reply:
Hmmm, I don't know why SSIL and SSAO are different here. AFAIK in both cases we should be inserting the barrier between blur_passes. Also, in both cases, there is only one pass when using very low quality. It seems the issue is that SSIL is inserting a barrier between each of the 4 sub passes, even though they have no dependencies on each other. SSIL should look the same as this one then, I.e. it should add a barrier after every blur_pass and should not add barriers within blur passes

Reply:
Yeah I approached this PR with the perspective of just fixing the error, but I haven't analyzed what the content does. The main problem behind the error just comes from the lack of a barrier between the different steps, which are missing in very low quality. Perhaps you'd rather approach this by just deleting these altogether and just insert a barrier after each step?